File: Backend/model/src/main/java/io/sparkled/model/validator/SongValidator.java
package io.sparkled.model.validator;

import io.sparkled.model.entity.Song;
import io.sparkled.model.validator.exception.EntityValidationException;

public class SongValidator {

    private Song song;

    public SongValidator(Song song) {
        this.song = song;
    }

    public void validate() {
        if (song.getName() == null) {
            throw new EntityValidationException("Song name must not be empty.");
        }
    }
}

File: model/src/main/kotlin/io/sparkled/model/validator/SongValidator.kt
package io.sparkled.model.validator

import io.sparkled.model.entity.Song
import io.sparkled.model.validator.exception.EntityValidationException

class SongValidator {

    fun validate(song: Song) {
        if (song.getName() == null) {
            throw EntityValidationException(Errors.NAME_MISSING)
        }
    }

    private object Errors {
        internal val NAME_MISSING = "Song name must not be empty."
    }
}
